README.md

```diff
@@ -2,13 +2,160 @@
 
 [![CircleCI](https://circleci.com/gh/Day8/re-frame-test.svg?style=svg)](https://circleci.com/gh/Day8/re-frame-test)
 
-A Clojure library designed to ... well, that part is up to you.
+`re-frame-test` is a library which provides helper functions for testing re-frame projects
 
+## Philosopy 
+When testing a re-frame application you often what to test the effects of events
+you have written on the state of the application.
 
-## Usage
+In general a re-frame test would
 
-FIXME
+ 1. run some test fixtures
+ 2. subscribe to some state
+ 3. assert the initial state
+ 4. dispatch an event
+ 5. assert the the state has changed
+ 6. reset the app state so there is no interaction between tests
+ 
+ 
+## Implementation
 
+`re-frame-test` provides two macros that together with `cljs.test` are useful to write tests with the above 
+philosophy. 
+ 
+### run-test-sync
+Execute `body` as a test, where each `dispatch` call is executed
+synchronously (via `dispatch-sync`), and any subsequent dispatches which are
+caused by that dispatch are also fully handled/executed prior to control flow
+returning to your test.
+
+Think of it kind of as though every `dispatch` in your app had been magically
+turned into `dispatch-sync`, and re-frame had lifted the restriction that says
+you can't call `dispatch-sync` from within an event handler.
+
+This macro is applicable for most events that do not run async behaviour within the 
+event.
+
+From the todomvc example:
+
+    (defn test-fixtures
+      []
+      ;; change this coeffect to make tests start with nothing
+      (rf/reg-cofx
+        :local-store-todos
+        (fn [cofx _]
+          "Read in todos from localstore, and process into a map we can merge into app-db."
+          (assoc cofx :local-store-todos
+                      (sorted-map)))))
+
+Define some test-fixtures in this case we have to ignore the localstore
+in the tests.
+
+    (deftest basic--sync
+      (rf-test/run-test-sync
+        (test-fixtures)
+        (rf/dispatch [:initialise-db])
+        
+Use the `run-test-sync` macro to construct the tests and initialise the app state
+note that the `dispatch` will be handled before the following code is executed, 
+effectively turning it into a `dispatch-sync`. Also any changes to the database
+and registrations will be rolled back at the termination of the test, therefore 
+our fixtures are run within the `run-test-sync` macro.
+    
+        (let [showing         (rf/subscribe [:showing])
+              sorted-todos    (rf/subscribe [:sorted-todos])
+              todos           (rf/subscribe [:todos])
+              visible-todos   (rf/subscribe [:visible-todos])
+              all-complete?   (rf/subscribe [:all-complete?])
+              completed-count (rf/subscribe [:completed-count])
+              footer-counts   (rf/subscribe [:footer-counts])]
+              
+Define subscriptions to the app state
```

Comment:
These comments should be Clojure comments next to the code so people can copy and read this as a single block of code.